packages/angular/src/generators/ng-add/utilities/lib.migrator.ts

```diff
@@ -0,0 +1,519 @@
+      );
+    } else {
+      this.updateBuildTargetConfiguration();
+      this.updateLintTargetConfiguration();
+      this.updateTestTargetConfiguration();
+    }
+
+    updateProjectConfiguration(this.tree, this.project.name, {
+      ...this.projectConfig,
+    });
+
+    await convertToNxProjectGenerator(this.tree, {
+      project: this.project.name,
+      skipFormat: true,
+    });
+  }
+
+  private updateNgPackageJson(): void {
+    const buildTarget = this.projectConfig.targets?.[this.targetNames.build];
+    if (
+      !buildTarget?.options?.project ||
+      !this.tree.exists(buildTarget.options.project)
+    ) {
+      // we already logged a warning for these cases, so just return
+      return;
+    }
+
+    const ngPackageJson = readJson(this.tree, buildTarget.options.project);
+    const offset = offsetFromRoot(this.project.newRoot);
+    ngPackageJson.$schema =
+      ngPackageJson.$schema &&
+      `${offset}node_modules/ng-packagr/ng-package.schema.json`;
+    ngPackageJson.dest = `${offset}dist/${this.project.name}`;
+    writeJson(this.tree, buildTarget.options.project, ngPackageJson);
+  }
+
+  private updateTsConfigs(): void {
+    const rootTsConfigFile = getRootTsConfigPathInTree(this.tree);
+    const projectOffsetFromRoot = offsetFromRoot(this.projectConfig.root);
+
+    this.updateTsConfigFileUsedByBuildTarget(
+      rootTsConfigFile,
+      projectOffsetFromRoot
+    );
+    this.updateTsConfigFileUsedByTestTarget(
+      rootTsConfigFile,
+      projectOffsetFromRoot
+    );
+  }
+
+  private updateEsLintConfig(): void {
+    if (!this.targetNames.lint || !this.tree.exists(this.newEsLintConfigPath)) {
+      return;
+    }
+
+    updateJson(this.tree, this.newEsLintConfigPath, (json) => {
+      delete json.root;
+      json.ignorePatterns = ['!**/*'];
+
+      const rootEsLintConfigRelativePath = joinPathFragments(
+        offsetFromRoot(this.projectConfig.root),
+        '.eslintrc.json'
+      );
+      if (Array.isArray(json.extends)) {
+        json.extends = json.extends.map((extend: string) =>
+          this.convertEsLintConfigExtendToNewPath(
+            this.oldEsLintConfigPath,
+            extend
+          )
+        );
+
+        // it might have not been extending from the root config, make sure it does
+        if (!json.extends.includes(rootEsLintConfigRelativePath)) {
+          json.extends.unshift(rootEsLintConfigRelativePath);
+        }
+      } else {
+        json.extends = rootEsLintConfigRelativePath;
+      }
+
+      json.overrides?.forEach((override) => {
+        if (!override.parserOptions?.project) {
+          return;
+        }
+
+        override.parserOptions.project = [
+          `${this.projectConfig.root}/tsconfig.*?.json`,
+        ];
+      });
+
+      return json;
+    });
+  }
+
+  private updateCacheableOperations(): void {
+    const workspaceConfig = readWorkspaceConfiguration(this.tree);
+
+    Object.keys(workspaceConfig.tasksRunnerOptions).forEach(
```

Comment:
I noticed elsewhere we were using ` ?? {}` on Object methods, should we do the same here?

Reply:
This one is guaranteed to be defined because we generate the `nx.json` with that option. That said, I'll add a check so the migrator is as independent as it can be from the global migration context and more resilient to changes.

Reply:
Sounds good!